src/spec/Reference.php

```diff
@@ -29,6 +29,9 @@
  */
 class Reference implements SpecObjectInterface, DocumentContextInterface
 {
+    /** @var array<string, mixed>  */
+    private static $relativeReferencesCache = [];
```

Comment:
this could probably cause issues when the library is used to read multiple different OpenAPI descriptions in the same process. Ideally the cache should be bound to the context like we have it with `ReferenceContextCache`.